core/pending.go

```diff
@@ -0,0 +1,112 @@
+package core
+
+import (
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+type PendingDataVariant uint8
+
+const (
+	PendingBlockVariant PendingDataVariant = iota + 1
+	PreConfirmedBlockVariant
+)
+
+type PendingDataInterface interface {
+	GetBlock() *Block
+	GetHeader() *Header
+	GetTransactions() []Transaction
+	GetStateUpdate() *StateUpdate
+	GetNewClasses() map[felt.Felt]Class
+	GetCandidateTransaction() []Transaction
+	Variant() PendingDataVariant
+}
+
+type PendingData struct {
+	data PendingDataInterface
+}
+
+func NewPendingData(data PendingDataInterface) *PendingData {
+	return &PendingData{
+		data: data,
+	}
+}
+
+func (p *PendingData) GetBlock() *Block {
+	return p.data.GetBlock()
+}
+
+func (p *PendingData) GetHeader() *Header {
+	return p.data.GetHeader()
+}
+
+func (p *PendingData) GetTransactions() []Transaction {
+	return p.data.GetTransactions()
+}
+
+func (p *PendingData) GetStateUpdate() *StateUpdate {
+	return p.data.GetStateUpdate()
+}
+
+func (p *PendingData) GetNewClasses() map[felt.Felt]Class {
+	return p.data.GetNewClasses()
+}
+
+func (p *PendingData) GetCandidateTransaction() []Transaction {
+	return p.data.GetCandidateTransaction()
+}
+
+func (p *PendingData) Variant() PendingDataVariant {
+	return p.data.Variant()
+}
+
+type PreConfirmed struct {
+	Block       *Block
+	StateUpdate *StateUpdate
+	// Does not fetch unknown classes. but we keep it for sequencer
+	NewClasses map[felt.Felt]Class
+	// We kept this field for sequencer, else non-sequencer node should use StateUpdate
+	TransactionStateDiffs []*StateDiff
+	CandidateTxs          []Transaction
+}
+
+func NewPreConfirmed(block *Block, stateUpdate *StateUpdate, transactionStateDiffs []*StateDiff, candidateTxs []Transaction) *PreConfirmed {
+	return &PreConfirmed{
+		Block:                 block,
+		StateUpdate:           stateUpdate,
+		NewClasses:            make(map[felt.Felt]Class),
+		TransactionStateDiffs: transactionStateDiffs,
+		CandidateTxs:          candidateTxs,
+	}
+}
+
+func (p *PreConfirmed) AsPendingData() *PendingData {
+	return &PendingData{data: p}
+}
```

Comment:
I believe you should return by value, it just holds a pointer. `PendingData` represents a pointer to `PreConfirmed` by returning a pointer to `PendingData` you are returning a pointer to a pointer (basically, `**PreConfirmed`). Is that the intention?